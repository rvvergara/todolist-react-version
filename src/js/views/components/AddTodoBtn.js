import React from 'react';
import PropTypes from 'prop-types';

const AddTodoBtn = (props) => {
  const {
    handleTodoBtn,
  } = props;

  return (
    <button
      type="button"
      className="btn btn-primary btn-block"
      onClick={handleTodoBtn}
    >
    Add Todo
    </button>
  );
};

AddTodoBtn.propTypes = {
  handleTodoBtn: PropTypes.func.isRequired,
};

export default AddTodoBtn;
