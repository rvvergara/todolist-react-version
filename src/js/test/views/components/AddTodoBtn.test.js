import React from 'react';
import { shallow } from 'enzyme';
import AddTodoBtn from '../../../views/components/AddTodoBtn';

describe('AddTodoBtn', () => {
  let wrapper;
  let handleTodoBtn;

  beforeEach(() => {
    handleTodoBtn = jest.fn();
    wrapper = shallow(
      <AddTodoBtn
        handleTodoBtn={handleTodoBtn}
      />,
    );
  });

  test('it should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('it should call handleTodoBtn upon button click', () => {
    wrapper.find('button').simulate('click');
    expect(handleTodoBtn).toHaveBeenCalled();
  });
});
