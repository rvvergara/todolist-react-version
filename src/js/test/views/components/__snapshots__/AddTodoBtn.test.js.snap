// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodoBtn it should render correctly 1`] = `
<button
  className="btn btn-primary btn-block"
  onClick={[MockFunction]}
  type="button"
>
  Add Todo
</button>
`;
